checkstyle/README.adoc

```diff
@@ -224,3 +224,19 @@ Documentation and further configuration for each check can be found https://chec
   </executions>
 </plugin>
 ----
+
+=== Use file filter BeforeExecutionExclusionFileFilter to exclude files from being processed by the utility
+Uses regular expression to match the file name to exclude. Possible to create multiple filters if one is not enough.
+
+Example configuration of the filter to exclude all Test directory files:
+
+.pom.xml
+[source,xml]
+----
+<module name="Checker">
+  <module name="BeforeExecutionExclusionFileFilter">
+    <property name="fileNamePattern"
+      value=".*[\\/]src[\\/]test[\\/].*$"/>
+  </module>
```

Comment:
Can multiple different patterns be given? I would prefer adding each file explicitly one by one so you can also have a comment why it is excluded. Otherwise this can be very easily turn into "lets ignore everything in src/" bulk rule

Reply:
It's possible to create several filters, that way you can add each file explicitly using multiple different patterns.

For example:
`<module name="Checker">
   <module name="BeforeExecutionExclusionFileFilter">
      <property name="fileNamePattern" value="module\-info\.java$"/>
   </module>
   <module name="BeforeExecutionExclusionFileFilter">
      <property name="fileNamePattern" value="Record\.java$"/>
   </module>
</module>`